int32 speed
---
string message